lib/url.js

```diff
@@ -266,11 +266,25 @@ api.removeDotSegments = path => {
   return output.join('/');
 };
 
+// TODO: time better isAbsolute/isRelative checks using full regexes:
+// http://jmrware.com/articles/2009/uri_regexp/URI_regex.html
+
 /**
  * Returns true if the given value is an absolute IRI, false if not.
+ * Note: this is a weak check.
  *
  * @param v the value to check.
  *
  * @return true if the value is an absolute IRI, false if not.
  */
-api.isAbsolute = v => types.isString(v) && v.indexOf(':') !== -1;
+api.isAbsolute = v => types.isString(v) && v.includes(':');
```

Comment:
To do this right, along with other URI processing, really requires full parsing based on RFC3986, but if it passes tests, you probably don't need to go that far. I have a full implementation in Ruby [here](https://github.com/ruby-rdf/rdf/blob/develop/lib/rdf/model/uri.rb), which is unencumbered. You might think that rdflib.js would too, but it seems light.

Reply:
This and the new isRelative() call are obviously not very robust.  We could (should?) easily craft tests that break the code.  Like make a relative IRI with a ':' in it.  I did add a comment above for a page with regexes.  Was trying to avoid adding too much to this PR.

Reply:
An absolute URI has a schema followed by a colon, so something like `/^\w+:/` might work.

Reply:
I changed it check via a proper scheme regex.